packages/rocketchat-livechat/server/models/Rooms.js

```diff
@@ -196,6 +196,19 @@ RocketChat.models.Rooms.changeAgentByRoomId = function(roomId, newAgent) {
 	this.update(query, update);
 };
 
+RocketChat.models.Rooms.removeAgentByRoomId = function(roomId) {
+	const query = {
+		_id: roomId
+	};
+	const update = {
+		$set: {
```

Comment:
You should use `$unset` to remove the property completely. You're setting it to null, so property still exists. So for queries for rooms having `servedBy` properties, it would falsely return that room also.